app/services/user_destroyer.rb

```diff
@@ -111,6 +113,16 @@ def destroy(user, opts = {})
 
   protected
 
+  def prepare_for_destroy(user)
+    PostAction.where(user_id: user.id).delete_all
+    UserAction.where(user_id: user.id).delete_all
```

Comment:
Minor change, have a look at the user model, there are 2 other columns to match, feel free to merge once you change that, change is good!